pandas/core/arrays/categorical.py

```diff
@@ -100,7 +100,15 @@ def f(self, other):
         if is_scalar(other):
             if other in self.categories:
                 i = self.categories.get_loc(other)
-                return getattr(self._codes, op)(i)
+                f = getattr(self._codes, op)
+                ret = f(i)
+
+                # check for NaN in self
+                na_mask = (self._codes == -1)
+                if na_mask.any():
+                    # In other series, the leads to False, so do that here too
```

Comment:
I know you copied this comment from above, but it is actually not fully clear to me what it wants to say, so would just leave it out here.

(or say something that comparison to NaN always gives False)

Reply:
Sounds good! Great coaching!